aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/DeleteHandler.java

```diff
@@ -94,4 +93,28 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                 )
                 .then(progress -> ProgressEvent.defaultSuccessHandler(null));
     }
+
+    @VisibleForTesting
+    static String decideSnapshotIdentifier(
+            final ResourceHandlerRequest<ResourceModel> request,
+            final DBInstance dbInstance
+    ) {
+        // CFN sends true or false for snapshotRequested based on the template's DeletionPolicy.
+        // However, CCAPI leaves it null. In that case, we take a snapshot, since that is CFN's default.
+        // Exception is that final snapshots are not allowed for read replicas and cluster instances.
+        boolean shouldAttemptToTakeSnapshot = BooleanUtils.isNotFalse(request.getSnapshotRequested());
+        boolean isReadReplica = StringUtils.isNotEmpty(dbInstance.readReplicaSourceDBInstanceIdentifier()) ||
+                StringUtils.isNotEmpty(dbInstance.readReplicaSourceDBClusterIdentifier());
```

Comment:
Can we create a new file for physical resource helpers like https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/blob/master/aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/util/ResourceModelHelper.java#L14